### PR TITLE
JAMES-2813 Bind SerializedTaskManager to CassandraEventStore

### PR DESCRIPTION
**Pull request by blackheaven**
It should be a bit more readable than #2553

### CODE REVIEW

File: server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java [hunk 1]
```diff
@@ -35,74 +33,51 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 
-class TaskSerializationTest {
+public class TaskSerializationTest {
```
Comment (rouazana) [inline server/task-json/src/test/java/org/apache/james/server/task/json/TaskSerializationTest.java]: is it needed?

File: server/task/src/main/scala/org/apache/james/task/eventsourcing/RecentTasksProjection.scala [hunk 2]
```diff
@@ -23,18 +23,24 @@ import java.util.concurrent.ConcurrentLinkedDeque
...
+  def add(taskId: TaskId): Unit
+}
+
+class MemoryRecentTasksProjection() extends RecentTasksProjection {
 
   import scala.jdk.CollectionConverters._
 
   private val tasks = new ConcurrentLinkedDeque[TaskId]
 
   def list(): List[TaskId] = tasks.asScala.toList
```
Comment (Arsnael) [inline server/task/src/main/scala/org/apache/james/task/eventsourcing/RecentTasksProjection.scala]: Shouldn't this be `override` then?

File: server/container/guice/guice-common/pom.xml [hunk 3]
```diff
@@ -37,6 +37,23 @@
             <artifactId>apache-mailet-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-backends-cassandra</artifactId>
```
Comment (Arsnael) [inline server/container/guice/guice-common/pom.xml]: Wrong order

File: server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TaskEventsDTO.scala [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+package org.apache.james.task.eventsourcing.cassandra
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.apache.james.eventsourcing.EventId
+import org.apache.james.eventsourcing.eventstore.cassandra.dto.EventDTO
+import org.apache.james.server.task.json.JsonTaskSerializer
+import org.apache.james.task.{Task, TaskId}
+import org.apache.james.task.eventsourcing._
+
+sealed abstract class TaskEventDTO(val getType: String, val getAggregate: String, val getEvent: Int) extends EventDTO {
```
Comment (Arsnael) [inline server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/TaskEventsDTO.scala]: I'm no Scala expert but is it normal the name of the file is slightly different with the name of the class?
`TaskEventsDTO` vs `TaskEventDTO` ? (`s` diff at Event)

File: server/task-cassandra/bin/pom.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    <name>Apache James :: Server :: Task</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-util</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Comment (trantienduchn) [inline server/task-cassandra/bin/pom.xml]: junit 5?

File: server/task-cassandra/bin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>james-server</artifactId>
+        <groupId>org.apache.james</groupId>
+        <version>3.2.0-SNAPSHOT</version>
```
Comment (trantienduchn) [inline server/task-cassandra/bin/pom.xml]: should it be `3.4.0-SNAPSHOT` ?

File: server/task-cassandra/bin/pom.xml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    </parent>
+
+    <artifactId>james-server-task</artifactId>
```
Comment (trantienduchn) [inline server/task-cassandra/bin/pom.xml]: I feel the artifactId doesn't match with its pom path! (`james-server-task` vs `server/task-cassandra/bin/pom.xml` )

File: server/task/src/main/scala/org/apache/james/task/eventsourcing/TaskExecutionDetailsProjection.scala [hunk 8]
```diff
@@ -39,15 +37,21 @@ class TaskExecutionDetailsProjection() {
...
     load(taskId)
       .map(updater)
       .foreach(update(taskId, _))
+
+  def load(taskId: TaskId): Option[TaskExecutionDetails]
+  def update(taskId: TaskId, details: TaskExecutionDetails): Unit
+}
+
+class MemoryTaskExecutionDetailsProjection() extends TaskExecutionDetailsProjection {
+  private[this] val projections = new ConcurrentHashMap[TaskId, TaskExecutionDetails]
```
Comment (chibenwa) [inline server/task/src/main/scala/org/apache/james/task/eventsourcing/TaskExecutionDetailsProjection.scala]: I would call this variable 'details'

File: server/container/guice/guice-common/pom.xml (hunk 3 above)
Comment (blackheaven) [reply server/container/guice/guice-common/pom.xml]: ?

File: server/task-cassandra/bin/pom.xml (hunk 5 above)
Comment (trantienduchn) [reply server/task-cassandra/bin/pom.xml]: why thumbdown?

File: server/task-cassandra/bin/pom.xml (hunk 6 above)
Comment (trantienduchn) [reply server/task-cassandra/bin/pom.xml]: why thumbdown?

File: server/task-cassandra/bin/pom.xml (hunk 7 above)
Comment (trantienduchn) [reply server/task-cassandra/bin/pom.xml]: why thumbdown?

Comment (blackheaven) [reply server/task-cassandra/bin/pom.xml]: it's in the `bin` directory, I have removed it.

File: server/container/guice/guice-common/pom.xml (hunk 3 above)
Comment (Arsnael) [reply server/container/guice/guice-common/pom.xml]: apache-`m`ailet-test
apache-`j`ames-backends-cassandra

`j` before `m`, wrong order :)

File: server/container/guice/guice-common/pom.xml [hunk 9]
```diff
@@ -139,6 +156,36 @@
...
+            <artifactId>james-server-task</artifactId>
+            <scope>test</scope>
+            <type>test-jar</type>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-task-cassandra</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
```
Comment (remk) [inline server/container/guice/guice-common/pom.xml]: no need to repeat it with scope test

File: server/container/guice/guice-common/pom.xml [hunk 10]
```diff
@@ -139,6 +156,36 @@
...
+            <artifactId>james-server-task-cassandra</artifactId>
```
Comment (remk) [inline server/container/guice/guice-common/pom.xml]: no need to repeat it with scope test

File: server/container/guice/guice-common/pom.xml [hunk 11]
```diff
@@ -139,6 +156,36 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-task-json</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-task-json</artifactId>
```
Comment (remk) [inline server/container/guice/guice-common/pom.xml]: no need to repeat it with scope test

File: server/container/guice/guice-common/pom.xml (hunk 3 above)
Comment (blackheaven) [reply server/container/guice/guice-common/pom.xml]: Ouch, I'm tired

**Comment by blackheaven** [comment]
```
[ERROR]   JamesMailSpoolerTest.threadSuicideShouldAckTheItem:126 » ConditionTimeout Cond...
```

test this please

**Comment by blackheaven** [comment]
```
[ERROR] Failed to execute goal on project james-server-task: Could not resolve dependencies for project org.apache.james:james-server-task:jar:3.4.0-SNAPSHOT: Could not transfer artifact org.scala-lang.modules:scala-java8-compat_2.12:jar:0.9.0 from/to central (https://repo.maven.apache.org/maven2): GET request of: org/scala-lang/modules/scala-java8-compat_2.12/0.9.0/scala-java8-compat_2.12-0.9.0.jar from central failed: Connection reset -> [Help 1]
```

test this please

**Comment by blackheaven** [comment]
```
[ERROR] Exit code: 1 - OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000c0000000, 179306496, 0) failed; error='Cannot allocate memory' (errno=12)
```

test this please

**Comment by Arsnael** [comment]
```
[4bd799e8914a552140359cec22b74446de3882f1] [ERROR] Errors: 
[4bd799e8914a552140359cec22b74446de3882f1] [ERROR]   BucketBlobStoreContract.saveConcurrentlyWithNonPreExistingBucketShouldNotFail:153 » NotTerminated
```
test this please

**Comment by chibenwa** [comment]
Please indicate the test class as well as the maven project in case of failure...

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**